packaging/os/zypper_repository.py

```diff
@@ -238,14 +264,16 @@ def get_zypper_version(module):
         return LooseVersion('1.0')
     return LooseVersion(stdout.split()[1])
 
+def runrefreshrepo(module, auto_import_keys=False, shortname=None):
+    "Forces zypper to refresh repo metadata."
+    cmd = ['/usr/bin/zypper', 'refresh', '--force']
+    if auto_import_keys:
+        cmd.append('--gpg-auto-import-keys')
+    if shortname is not None:
+        cmd.extend(['-r', shortname])
 
-def fail_if_rc_is_null(module, rc, stdout, stderr):
-    if rc != 0:
-        #module.fail_json(msg=stderr if stderr else stdout)
-        if stderr:
-            module.fail_json(msg=stderr)
-        else:
-            module.fail_json(msg=stdout)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+    return rc, stdout, stderr
```

Comment:
The idea was to make each of the functions self-documenting, by naming the return values.
For the next refactor it's probably a good idea to move the run_command to the main function and only build the command strings in the individual ones. Then one could also better deal with multiple return codes.

Thanks for the review!